// @vapor-ui/core/dist/components/text-input/index.js
'use client';
import "./index.css";
var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};

// src/components/text-input/text-input.tsx
import { forwardRef, useId } from "react";
import { Primitive } from "@radix-ui/react-primitive";
import clsx from "clsx";

// src/libs/create-context.ts
import { createContext as createReactContext, useContext as useReactContext } from "react";
var getErrorMessage = (hook, provider) => {
  return `${hook} returned \`undefined\`. Seems you forgot to wrap component within ${provider}`;
};
var createContext = ({
  name,
  strict = true,
  hookName = "useContext",
  providerName = "Provider",
  errorMessage,
  defaultValue
} = {}) => {
  const Context = createReactContext(defaultValue);
  Context.displayName = name;
  function useContext() {
    const context = useReactContext(Context);
    if (!context && strict) {
      const error = new Error(errorMessage != null ? errorMessage : getErrorMessage(hookName, providerName));
      error.name = "ContextError";
      throw error;
    }
    return context;
  }
  return [Context.Provider, useContext, Context];
};

// src/utils/create-split-props.ts
var createSplitProps = () => (props, keys) => keys.reduce(
  (previousValue, currentValue) => {
    const [target, source] = previousValue;
    const key = currentValue;
    if (source[key] !== void 0) {
      target[key] = source[key];
    }
    delete source[key];
    return [target, source];
  },
  [{}, __spreadValues({}, props)]
);

// src/components/text-input/text-input.css.ts
import { createRuntimeFn as _7a468 } from "@vanilla-extract/recipes/createRuntimeFn";
var field = _7a468({ defaultClassName: "interactions__13di4x95 interactions__13di4x91 interactions-scale_normal__13di4x96 interactions__13di4x94 text-input__npu94q5", variantClassNames: { invalid: { true: "text-input-invalid_true__npu94q7" }, size: { sm: "text-input-size_sm__npu94q8", md: "text-input-size_md__npu94q9", lg: "text-input-size_lg__npu94qa", xl: "text-input-size_xl__npu94qb" } }, defaultVariants: { invalid: false, size: "md" }, compoundVariants: [] });
var label = _7a468({ defaultClassName: "text-input__npu94q3 text-input__npu94q2", variantClassNames: { visuallyHidden: { true: "visually-hidden__1q7xssa0" } }, defaultVariants: { visuallyHidden: false }, compoundVariants: [] });
var root = _7a468({ defaultClassName: "text-input__npu94q0", variantClassNames: { disabled: { true: "text-input-disabled_true__npu94q1" } }, defaultVariants: { disabled: false }, compoundVariants: [] });

// src/components/text-input/text-input.tsx
import { jsx } from "react/jsx-runtime";
var [TextInputProvider, useTextInputContext] = createContext({
  name: "TextInput",
  hookName: "useTextInputContext",
  providerName: "TextInputProvider"
});
var Root = forwardRef(
  (_a, ref) => {
    var _b = _a, { className, children } = _b, props = __objRest(_b, ["className", "children"]);
    const textInputId = useId();
    const [textInputRootProps, otherProps] = createSplitProps()(props, [
      "type",
      "value",
      "onValueChange",
      "defaultValue",
      "size",
      "disabled",
      "invalid",
      "readOnly",
      "visuallyHidden",
      "placeholder"
    ]);
    const { disabled } = textInputRootProps;
    return /* @__PURE__ */ jsx(TextInputProvider, { value: __spreadValues({ textInputId }, textInputRootProps), children: /* @__PURE__ */ jsx(
      Primitive.div,
      __spreadProps(__spreadValues({
        ref,
        className: clsx(root({ disabled }), className)
      }, otherProps), {
        children
      })
    ) });
  }
);
Root.displayName = "TextInput.Root";
var Label = forwardRef(
  (_a, ref) => {
    var _b = _a, { htmlFor, className } = _b, props = __objRest(_b, ["htmlFor", "className"]);
    const { textInputId = htmlFor, visuallyHidden } = useTextInputContext();
    return /* @__PURE__ */ jsx(
      Primitive.label,
      __spreadValues({
        ref,
        htmlFor: textInputId,
        className: clsx(label({ visuallyHidden }), className)
      }, props)
    );
  }
);
Label.displayName = "TextInput.Label";
var Field = forwardRef(
  (_a, ref) => {
    var _b = _a, { id, className } = _b, props = __objRest(_b, ["id", "className"]);
    const {
      type,
      textInputId = id,
      value,
      onValueChange,
      defaultValue,
      disabled,
      invalid,
      readOnly,
      size,
      placeholder
    } = useTextInputContext();
    return /* @__PURE__ */ jsx(
      Primitive.input,
      __spreadValues({
        ref,
        id: textInputId,
        type,
        value,
        onChange: (event) => onValueChange == null ? void 0 : onValueChange(event.target.value),
        defaultValue,
        disabled,
        "aria-invalid": invalid,
        readOnly,
        placeholder,
        className: clsx(field({ invalid, size }), className)
      }, props)
    );
  }
);
Field.displayName = "TextInput.Field";
var TextInput = { Root, Label, Field };
export {
  TextInput,
  Field as TextInputField,
  Label as TextInputLabel,
  Root as TextInputRoot
};
