// @vapor-ui/core/dist/components/radio-group/index.cjs
'use client';
"use strict";
require("./index.css");
var __create = Object.create;
var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/components/radio-group/index.ts
var radio_group_exports = {};
__export(radio_group_exports, {
  RadioGroup: () => RadioGroup,
  RadioGroupControl: () => Control,
  RadioGroupItem: () => Item,
  RadioGroupLabel: () => Label,
  RadioGroupRoot: () => Root
});
module.exports = __toCommonJS(radio_group_exports);

// src/components/radio-group/radio-group.tsx
var import_react2 = require("react");
var import_react_primitive = require("@radix-ui/react-primitive");
var import_react_radio_group = require("@radix-ui/react-radio-group");
var import_clsx = __toESM(require("clsx"));

// src/libs/create-context.ts
var import_react = require("react");
var getErrorMessage = (hook, provider) => {
  return `${hook} returned \`undefined\`. Seems you forgot to wrap component within ${provider}`;
};
var createContext = ({
  name,
  strict = true,
  hookName = "useContext",
  providerName = "Provider",
  errorMessage,
  defaultValue
} = {}) => {
  const Context = (0, import_react.createContext)(defaultValue);
  Context.displayName = name;
  function useContext() {
    const context = (0, import_react.useContext)(Context);
    if (!context && strict) {
      const error = new Error(errorMessage != null ? errorMessage : getErrorMessage(hookName, providerName));
      error.name = "ContextError";
      throw error;
    }
    return context;
  }
  return [Context.Provider, useContext, Context];
};

// src/utils/create-split-props.ts
var createSplitProps = () => (props, keys) => keys.reduce(
  (previousValue, currentValue) => {
    const [target, source] = previousValue;
    const key = currentValue;
    if (source[key] !== void 0) {
      target[key] = source[key];
    }
    delete source[key];
    return [target, source];
  },
  [{}, __spreadValues({}, props)]
);

// src/components/radio-group/radio-group.css.ts
var import_createRuntimeFn = require("@vanilla-extract/recipes/createRuntimeFn");
var control = (0, import_createRuntimeFn.createRuntimeFn)({ defaultClassName: "interactions__13di4x95 interactions__13di4x91 interactions-scale_normal__13di4x96 interactions__13di4x93 radio-group__2zibu7e", variantClassNames: { invalid: { true: "radio-group__2zibu7f" }, size: { md: "radio-group__2zibu7g", lg: "radio-group__2zibu7h" } }, defaultVariants: { invalid: false, size: "md" }, compoundVariants: [] });
var indicator = "radio-group__2zibu7m";
var item = (0, import_createRuntimeFn.createRuntimeFn)({ defaultClassName: "radio-group__2zibu7c radio-group__2zibu7a", variantClassNames: { disabled: { true: "radio-group__2zibu7b" } }, defaultVariants: { disabled: false }, compoundVariants: [] });
var label = (0, import_createRuntimeFn.createRuntimeFn)({ defaultClassName: "radio-group__2zibu7o radio-group__2zibu7n", variantClassNames: { visuallyHidden: { true: "visually-hidden__1q7xssa0" } }, defaultVariants: { visuallyHidden: false }, compoundVariants: [] });
var root = (0, import_createRuntimeFn.createRuntimeFn)({ defaultClassName: "radio-group__2zibu75 radio-group__2zibu70", variantClassNames: { size: { md: "radio-group__2zibu71", lg: "radio-group__2zibu72" }, orientation: { horizontal: "radio-group__2zibu73", vertical: "radio-group__2zibu74" } }, defaultVariants: { orientation: "vertical" }, compoundVariants: [] });

// src/components/radio-group/radio-group.tsx
var import_jsx_runtime = require("react/jsx-runtime");
var [RadioGroupProvider, useRadioGroupContext] = createContext({
  name: "RadioGroup",
  hookName: "useRadioGroupContext",
  providerName: "RadioGroupProvider"
});
var Root = (0, import_react2.forwardRef)((_a, ref) => {
  var _b = _a, { className } = _b, props = __objRest(_b, ["className"]);
  const [sharedProps, otherProps] = createSplitProps()(props, [
    "name",
    "required",
    "disabled",
    "value",
    "onValueChange",
    "defaultValue",
    "dir",
    "loop",
    "orientation",
    "invalid",
    "size",
    "visuallyHidden"
  ]);
  const { size, orientation } = sharedProps;
  return /* @__PURE__ */ (0, import_jsx_runtime.jsx)(RadioGroupProvider, { value: sharedProps, children: /* @__PURE__ */ (0, import_jsx_runtime.jsx)(
    import_react_radio_group.Root,
    __spreadValues(__spreadValues({
      ref,
      className: (0, import_clsx.default)(root({ size, orientation }), className)
    }, sharedProps), otherProps)
  ) });
});
Root.displayName = "RadioGroup.Root";
var [RadioGroupItemProvider, useRadioGroupItemContext] = createContext({
  name: "RadioGroupItem",
  hookName: "useRadioGroupItemContext",
  providerName: "RadioGroupItemProvider"
});
var Item = (0, import_react2.forwardRef)((_a, ref) => {
  var _b = _a, { className } = _b, props = __objRest(_b, ["className"]);
  const radioGroupItemId = (0, import_react2.useId)();
  const rootContext = useRadioGroupContext();
  const [itemProps, otherProps] = createSplitProps()(props, [
    "value",
    "disabled",
    "visuallyHidden",
    "size",
    "invalid"
  ]);
  const {
    disabled = rootContext.disabled,
    invalid = rootContext.invalid,
    visuallyHidden = rootContext.visuallyHidden
  } = itemProps;
  return /* @__PURE__ */ (0, import_jsx_runtime.jsx)(
    RadioGroupItemProvider,
    {
      value: __spreadProps(__spreadValues({}, itemProps), { radioGroupItemId, disabled, invalid, visuallyHidden }),
      children: /* @__PURE__ */ (0, import_jsx_runtime.jsx)(
        import_react_primitive.Primitive.div,
        __spreadValues({
          ref,
          className: (0, import_clsx.default)(item({ disabled }), className)
        }, otherProps)
      )
    }
  );
});
Item.displayName = "RadioGroup.Item";
var Control = (0, import_react2.forwardRef)(
  (_a, ref) => {
    var _b = _a, { id, className } = _b, props = __objRest(_b, ["id", "className"]);
    const { size } = useRadioGroupContext();
    const { radioGroupItemId, value, invalid, disabled } = useRadioGroupItemContext();
    return /* @__PURE__ */ (0, import_jsx_runtime.jsx)(
      import_react_radio_group.Item,
      __spreadProps(__spreadValues({
        ref,
        id: id || radioGroupItemId,
        value,
        disabled,
        "aria-invalid": invalid,
        className: (0, import_clsx.default)(control({ size, invalid }), className)
      }, props), {
        children: /* @__PURE__ */ (0, import_jsx_runtime.jsx)(import_react_radio_group.Indicator, { className: (0, import_clsx.default)(indicator) })
      })
    );
  }
);
Control.displayName = "RadioGroup.Control";
var Label = (0, import_react2.forwardRef)(
  (_a, ref) => {
    var _b = _a, { htmlFor, className } = _b, props = __objRest(_b, ["htmlFor", "className"]);
    const { radioGroupItemId, visuallyHidden } = useRadioGroupItemContext();
    return /* @__PURE__ */ (0, import_jsx_runtime.jsx)(
      import_react_primitive.Primitive.label,
      __spreadValues({
        ref,
        htmlFor: htmlFor || radioGroupItemId,
        className: (0, import_clsx.default)(label({ visuallyHidden }), className)
      }, props)
    );
  }
);
Label.displayName = "RadioGroup.Label";
var RadioGroup = { Root, Item, Control, Label };
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  RadioGroup,
  RadioGroupControl,
  RadioGroupItem,
  RadioGroupLabel,
  RadioGroupRoot
});
