// @vapor-ui/core/dist/components/create-theme-config/index.cjs
"use strict";
var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/components/create-theme-config/index.ts
var create_theme_config_exports = {};
__export(create_theme_config_exports, {
  createThemeConfig: () => createThemeConfig
});
module.exports = __toCommonJS(create_theme_config_exports);

// src/components/theme-inject/theme-injector.ts
var DARK_CLASS_NAME = "vapor-dark-theme";
var LIGHT_CLASS_NAME = "vapor-light-theme";
var THEME_CONFIG = {
  STORAGE_KEY: "vapor-ui-theme",
  CLASS_NAMES: {
    dark: DARK_CLASS_NAME,
    light: LIGHT_CLASS_NAME
  },
  RADIUS_FACTOR_MAP: {
    none: 0,
    sm: 0.5,
    md: 1,
    lg: 1.5,
    xl: 2,
    full: 3
  }
};

// src/components/create-theme-config/create-theme-config.ts
var DEFAULT_THEME = {
  appearance: "light",
  radius: "md",
  scaling: 1
};
var createThemeConfig = (userConfig) => {
  const _a = userConfig != null ? userConfig : {}, {
    storageKey = THEME_CONFIG.STORAGE_KEY,
    nonce,
    enableSystemTheme = false
  } = _a, themeProps = __objRest(_a, [
    "storageKey",
    "nonce",
    "enableSystemTheme"
  ]);
  return __spreadProps(__spreadValues(__spreadValues({}, DEFAULT_THEME), themeProps), {
    storageKey,
    nonce,
    enableSystemTheme
  });
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  createThemeConfig
});
