// @vapor-ui/core/dist/components/checkbox/index.cjs
'use client';
"use strict";
require("./index.css");
var __create = Object.create;
var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/components/checkbox/index.ts
var checkbox_exports = {};
__export(checkbox_exports, {
  Checkbox: () => Checkbox,
  CheckboxControl: () => Control,
  CheckboxLabel: () => Label,
  CheckboxRoot: () => Root
});
module.exports = __toCommonJS(checkbox_exports);

// src/components/checkbox/checkbox.tsx
var import_react2 = require("react");
var import_react_checkbox = require("@radix-ui/react-checkbox");
var import_react_primitive = require("@radix-ui/react-primitive");
var import_react_use_controllable_state = require("@radix-ui/react-use-controllable-state");
var import_clsx = __toESM(require("clsx"));

// src/libs/create-context.ts
var import_react = require("react");
var getErrorMessage = (hook, provider) => {
  return `${hook} returned \`undefined\`. Seems you forgot to wrap component within ${provider}`;
};
var createContext = ({
  name,
  strict = true,
  hookName = "useContext",
  providerName = "Provider",
  errorMessage,
  defaultValue
} = {}) => {
  const Context = (0, import_react.createContext)(defaultValue);
  Context.displayName = name;
  function useContext() {
    const context = (0, import_react.useContext)(Context);
    if (!context && strict) {
      const error = new Error(errorMessage != null ? errorMessage : getErrorMessage(hookName, providerName));
      error.name = "ContextError";
      throw error;
    }
    return context;
  }
  return [Context.Provider, useContext, Context];
};

// src/utils/create-split-props.ts
var createSplitProps = () => (props, keys) => keys.reduce(
  (previousValue, currentValue) => {
    const [target, source] = previousValue;
    const key = currentValue;
    if (source[key] !== void 0) {
      target[key] = source[key];
    }
    delete source[key];
    return [target, source];
  },
  [{}, __spreadValues({}, props)]
);

// src/components/checkbox/checkbox.css.ts
var import_createRuntimeFn = require("@vanilla-extract/recipes/createRuntimeFn");
var control = (0, import_createRuntimeFn.createRuntimeFn)({ defaultClassName: "interactions__13di4x95 interactions__13di4x91 interactions-scale_normal__13di4x96 interactions__13di4x93 checkbox__a14mdw7", variantClassNames: { invalid: { true: "checkbox__a14mdw8" }, size: { md: "checkbox__a14mdw9", lg: "checkbox__a14mdwa" } }, defaultVariants: { invalid: false, size: "md" }, compoundVariants: [] });
var indicator = (0, import_createRuntimeFn.createRuntimeFn)({ defaultClassName: "checkbox__a14mdwi checkbox__a14mdwf", variantClassNames: { size: { md: "checkbox__a14mdwg", lg: "checkbox__a14mdwh" } }, defaultVariants: { size: "md" }, compoundVariants: [] });
var label = (0, import_createRuntimeFn.createRuntimeFn)({ defaultClassName: "typography__1saa0cgi typography__1saa0cgd checkbox__a14mdw4", variantClassNames: { visuallyHidden: { true: "visually-hidden__1q7xssa0" } }, defaultVariants: { visuallyHidden: false }, compoundVariants: [] });
var root = (0, import_createRuntimeFn.createRuntimeFn)({ defaultClassName: "checkbox__a14mdw2 checkbox__a14mdw0", variantClassNames: { disabled: { true: "checkbox__a14mdw1" } }, defaultVariants: { disabled: false }, compoundVariants: [] });

// src/components/checkbox/checkbox.tsx
var import_jsx_runtime = require("react/jsx-runtime");
var [CheckboxProvider, useCheckboxContext] = createContext({
  name: "Checkbox",
  hookName: "useCheckbox",
  providerName: "CheckboxProvider"
});
var Root = (0, import_react2.forwardRef)((_a, ref) => {
  var _b = _a, { className } = _b, props = __objRest(_b, ["className"]);
  const checkboxId = (0, import_react2.useId)();
  const [checkboxProps, otherProps] = createSplitProps()(props, [
    "checked",
    "onCheckedChange",
    "defaultChecked",
    "indeterminate",
    "size",
    "invalid",
    "disabled",
    "visuallyHidden"
  ]);
  const { disabled } = checkboxProps;
  return /* @__PURE__ */ (0, import_jsx_runtime.jsx)(CheckboxProvider, { value: __spreadValues({ checkboxId }, checkboxProps), children: /* @__PURE__ */ (0, import_jsx_runtime.jsx)(
    import_react_primitive.Primitive.div,
    __spreadValues({
      ref,
      className: (0, import_clsx.default)(root({ disabled }), className)
    }, otherProps)
  ) });
});
Root.displayName = "Checkbox.Root";
var Label = (0, import_react2.forwardRef)(
  (_a, ref) => {
    var _b = _a, { htmlFor, className } = _b, props = __objRest(_b, ["htmlFor", "className"]);
    const { checkboxId, visuallyHidden } = useCheckboxContext();
    return /* @__PURE__ */ (0, import_jsx_runtime.jsx)(
      import_react_primitive.Primitive.label,
      __spreadValues({
        ref,
        htmlFor: htmlFor || checkboxId,
        className: (0, import_clsx.default)(label({ visuallyHidden }), className)
      }, props)
    );
  }
);
var Control = (0, import_react2.forwardRef)(
  (_a, ref) => {
    var _b = _a, { id, className } = _b, props = __objRest(_b, ["id", "className"]);
    const {
      checkboxId,
      checked,
      onCheckedChange,
      defaultChecked,
      indeterminate,
      invalid,
      disabled,
      size
    } = useCheckboxContext();
    const [checkedState, setCheckedState] = (0, import_react_use_controllable_state.useControllableState)({
      prop: indeterminate ? "indeterminate" : checked,
      defaultProp: indeterminate ? "indeterminate" : defaultChecked || false,
      onChange: (state) => {
        if (state === "indeterminate") return;
        onCheckedChange == null ? void 0 : onCheckedChange(state);
      }
    });
    return /* @__PURE__ */ (0, import_jsx_runtime.jsx)(
      import_react_checkbox.Root,
      __spreadProps(__spreadValues({
        ref,
        id: checkboxId || id,
        checked: checkedState,
        onCheckedChange: setCheckedState,
        disabled,
        "aria-invalid": invalid,
        className: (0, import_clsx.default)(control({ invalid, size }), className)
      }, props), {
        children: /* @__PURE__ */ (0, import_jsx_runtime.jsxs)(import_react_checkbox.Indicator, { className: indicator({ size }), children: [
          checkedState === "indeterminate" && /* @__PURE__ */ (0, import_jsx_runtime.jsx)(DashIcon, {}),
          checkedState === true && /* @__PURE__ */ (0, import_jsx_runtime.jsx)(CheckIcon, {})
        ] })
      })
    );
  }
);
Control.displayName = "Checkbox.Control";
var CheckIcon = (props) => {
  return /* @__PURE__ */ (0, import_jsx_runtime.jsx)("svg", __spreadProps(__spreadValues({ viewBox: "0 0 8 7", xmlns: "http://www.w3.org/2000/svg" }, props), { children: /* @__PURE__ */ (0, import_jsx_runtime.jsx)(
    "path",
    {
      d: "M11.3135 5.29325c-.391-.391-1.024-.391-1.414 0l-3.364 3.364-.829-.828c-.39-.391-1.023-.391-1.414 0-.39.39-.39 1.023 0 1.414l1.536 1.535c.39.391 1.023.391 1.414 0l4.071-4.071c.391-.39.391-1.023 0-1.414",
      fill: "currentColor",
      fillRule: "evenodd",
      transform: "translate(-4 -5)"
    }
  ) }));
};
var DashIcon = (props) => {
  return /* @__PURE__ */ (0, import_jsx_runtime.jsx)("svg", __spreadProps(__spreadValues({ viewBox: "0 0 8 2", xmlns: "http://www.w3.org/2000/svg" }, props), { children: /* @__PURE__ */ (0, import_jsx_runtime.jsx)("rect", { fill: "currentColor", fillRule: "evenodd", height: "2", rx: "1", width: "8" }) }));
};
var Checkbox = { Root, Label, Control };
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  Checkbox,
  CheckboxControl,
  CheckboxLabel,
  CheckboxRoot
});
