// @vapor-ui/core/dist/components/nav/index.js
'use client';
import "./index.css";
var __defProp = Object.defineProperty;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};

// src/components/nav/nav.tsx
import { forwardRef } from "react";
import {
  Item as RadixItem,
  Link as RadixLink,
  List as RadixList,
  Root as RadixRoot
} from "@radix-ui/react-navigation-menu";
import clsx from "clsx";

// src/libs/create-context.ts
import { createContext as createReactContext, useContext as useReactContext } from "react";
var getErrorMessage = (hook, provider) => {
  return `${hook} returned \`undefined\`. Seems you forgot to wrap component within ${provider}`;
};
var createContext = ({
  name,
  strict = true,
  hookName = "useContext",
  providerName = "Provider",
  errorMessage,
  defaultValue
} = {}) => {
  const Context = createReactContext(defaultValue);
  Context.displayName = name;
  function useContext() {
    const context = useReactContext(Context);
    if (!context && strict) {
      const error = new Error(errorMessage != null ? errorMessage : getErrorMessage(hookName, providerName));
      error.name = "ContextError";
      throw error;
    }
    return context;
  }
  return [Context.Provider, useContext, Context];
};

// src/utils/create-split-props.ts
var createSplitProps = () => (props, keys) => keys.reduce(
  (previousValue, currentValue) => {
    const [target, source] = previousValue;
    const key = currentValue;
    if (source[key] !== void 0) {
      target[key] = source[key];
    }
    delete source[key];
    return [target, source];
  },
  [{}, __spreadValues({}, props)]
);

// src/components/nav/nav.css.ts
import { createRuntimeFn as _7a468 } from "@vanilla-extract/recipes/createRuntimeFn";
var item = _7a468({ defaultClassName: "nav__xuyk5bb", variantClassNames: { stretch: { true: "nav__xuyk5ba" } }, defaultVariants: {}, compoundVariants: [] });
var link = _7a468({ defaultClassName: "interactions__13di4x95 interactions__13di4x91 interactions-scale_normal__13di4x96 interactions__13di4x93 nav__xuyk5bd", variantClassNames: { disabled: { true: "nav__xuyk5be" }, align: { start: "nav__xuyk5bf", center: "nav__xuyk5bg", end: "nav__xuyk5bh" }, shape: { fill: "nav__xuyk5bi", ghost: "nav__xuyk5bj" }, size: { sm: "nav__xuyk5bk", md: "nav__xuyk5bl", lg: "nav__xuyk5bm", xl: "nav__xuyk5bn" } }, defaultVariants: { size: "md", shape: "fill", align: "center", disabled: false }, compoundVariants: [] });
var list = _7a468({ defaultClassName: "nav__xuyk5b5 nav__xuyk5b0", variantClassNames: { stretch: { true: "nav__xuyk5b1", false: "nav__xuyk5b2" }, direction: { horizontal: "nav__xuyk5b3", vertical: "nav__xuyk5b4" } }, defaultVariants: { direction: "horizontal", stretch: false }, compoundVariants: [] });

// src/components/nav/nav.tsx
import { jsx } from "react/jsx-runtime";
var [NavProvider, useNavContext] = createContext({
  name: "NavContext",
  providerName: "NavProvider",
  hookName: "useNavContext"
});
var Root = forwardRef((_a, ref) => {
  var _b = _a, { "aria-label": ariaLabel } = _b, props = __objRest(_b, ["aria-label"]);
  const [variantProps, otherProps] = createSplitProps()(props, [
    "direction",
    "size",
    "shape",
    "stretch",
    "align",
    "disabled"
  ]);
  const { direction } = variantProps;
  return /* @__PURE__ */ jsx(NavProvider, { value: variantProps, children: /* @__PURE__ */ jsx(
    RadixRoot,
    __spreadValues({
      ref,
      orientation: direction,
      "aria-label": ariaLabel || void 0
    }, otherProps)
  ) });
});
Root.displayName = "Nav.Root";
var List = forwardRef((_a, ref) => {
  var _b = _a, { className } = _b, props = __objRest(_b, ["className"]);
  const { direction, stretch } = useNavContext();
  return /* @__PURE__ */ jsx(
    RadixList,
    __spreadValues({
      ref,
      className: clsx(list({ direction, stretch }), className)
    }, props)
  );
});
List.displayName = "Nav.List";
var Item = forwardRef((_a, ref) => {
  var _b = _a, { className } = _b, props = __objRest(_b, ["className"]);
  const { stretch } = useNavContext();
  return /* @__PURE__ */ jsx(RadixItem, __spreadValues({ ref, className: clsx(item({ stretch }), className) }, props));
});
Item.displayName = "Nav.Item";
var Link = forwardRef(
  (_a, ref) => {
    var _b = _a, { selected, disabled, href, className } = _b, props = __objRest(_b, ["selected", "disabled", "href", "className"]);
    const { shape, size, align } = useNavContext();
    return /* @__PURE__ */ jsx(
      RadixLink,
      __spreadValues({
        ref,
        href: disabled ? void 0 : href,
        active: selected,
        "data-active": void 0,
        "data-selected": selected ? "true" : void 0,
        "aria-disabled": disabled ? "true" : void 0,
        className: clsx(link({ shape, size, align, disabled }), className)
      }, props)
    );
  }
);
Link.displayName = "Nav.Link";
var LinkItem = forwardRef((props, ref) => {
  return /* @__PURE__ */ jsx(Nav.Item, { children: /* @__PURE__ */ jsx(Nav.Link, __spreadValues({ ref }, props)) });
});
LinkItem.displayName = "Nav.LinkItem";
var Nav = { Root, List, Item, Link, LinkItem };
export {
  Nav,
  Item as NavItem,
  Link as NavLink,
  LinkItem as NavLinkItem,
  List as NavList,
  Root as NavRoot
};
